[fix] error docs + handle expected error (#7566)
sveltejs/kit: documentation/docs/20-core-concepts/10-routing.md

SvelteKit will 'walk up the tree' looking for the closest error boundary — if the file above didn't exist it would try `src/routes/blog/+error.svelte` and `src/routes/+error.svelte` before rendering the default error page. If _that_ fails (or if the error was thrown from the `load` function of the root `+layout`, which sits 'above' the root `+error`), SvelteKit will bail out and render a static fallback error page, which you can customise by creating a `src/error.html` file.

> > `+error.svelte` is _not_ used when an error occurs inside the [handle hook](/docs/hooks#server-hooks-handle) or an [endpoint](#server).

Reviewer: ```suggestion
> `+error.svelte` is _not_ used when an error occurs inside [`handle`](/docs/hooks#server-hooks-handle) or a [+server.js](#server) request handler.
```

The first argument to `Response` can be a [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream), making it possible to stream large amounts of data or create server-sent events (unless deploying to platforms that buffer responses, like AWS Lambda).

> You can use the `error`, `redirect` and `json` methods from `@sveltejs/kit` for convenience (but you don't have to). Note that `throw error(..)` (or an unexpected error) only returns a plain text error response, [+error.svelte](#error) is not used in this case as an endpoint doesn't live in the context of a page.

Reviewer: 'Plain text' is probably a bit misleading here — it's an HTML response that uses `src/error.html`, unless the `accept` header prefers JSON

```suggestion
You can use the `error`, `redirect` and `json` methods from `@sveltejs/kit` for convenience (but you don't have to).

If an error is thrown (either `throw error(...)` or an unexpected error), the response will be a JSON representation of the error or a fallback error page — which can be customised via `src/error.html` — depending on the `Accept` header. The [`+error.svelte`](#error) component will not be rendered.
```

I actually think we _might_ need a dedicated section on error handling, the `handleError` docs only get us so far — we gloss over `src/error.html` a bit. Need to think about how to do that without being duplicative
